### PR TITLE
[Perf] torch compile for dit and rope kernel

### PR DESCRIPTION
**Pull request by ZJY0516**
## Test Result
```
python text_to_image.py --model Tongyi-MAI/Z-Image-Turbo --num_inference_steps 9

Generation Configuration:
  Model: Tongyi-MAI/Z-Image-Turbo
  Inference steps: 9
  Cache backend: None (no acceleration)
  Parallel configuration: ulysses_degree=1
  Image size: 1024x1024
```
| z-image        | Time (s) |
|----------------|----------|
| PR             | 4.5626   |
| w/o torch compile | 4.6430   |
| main           | 5.4158   |

```
Generation Configuration:
  Model: Qwen/Qwen-Image
  Inference steps: 50
  Cache backend: None (no acceleration)
  Parallel configuration: ulysses_degree=1
  Image size: 1024x1024
```

| Qwen-Image        | Time (s) |
|----------------|----------|
| PR             | 61.4890   |
| w/o torch compile | 64.7892   |
| main           | 65.8851   |

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/diffusion_engine.py [hunk 1]
```diff
@@ -151,6 +157,22 @@ def _launch_workers(self, broadcast_handle):
     def add_req_and_wait_for_response(self, requests: list[OmniDiffusionRequest]):
         return scheduler.add_req(requests)
 
+    def _dummy_run(self):
+        """A dummy run to warm up the model."""
+        prompt = "dummy run"
+        num_inference_steps = 1
+        height = 1024
```
Comment (SamitHuang) [inline vllm_omni/diffusion/diffusion_engine.py:164]: why fix to 1024?

File: vllm_omni/diffusion/layers/rope.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import torch
+import triton
```
Comment (SamitHuang) [inline vllm_omni/diffusion/layers/rope.py]: do we need to add this dependency in pyproject?

File: vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py [hunk 3]
```diff
@@ -347,6 +307,7 @@ def forward(
         return img_attn_output, txt_attn_output
 
 
+@dit_support_compile()
```
Comment (SamitHuang) [inline vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py]: should we specify the dynamic dim to handle requests with different output image sizes? like the sequence dim

File: vllm_omni/diffusion/layers/rope.py (hunk 2 above)
Comment (ZJY0516) [reply vllm_omni/diffusion/layers/rope.py]: we will install triton automatically when we install vllm on cuda platform

File: vllm_omni/diffusion/diffusion_engine.py (hunk 1 above)
Comment (ZJY0516) [reply vllm_omni/diffusion/diffusion_engine.py:164]: Just a random number

File: vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py (hunk 3 above)
Comment (ZJY0516) [reply vllm_omni/diffusion/models/qwen_image/qwen_image_transformer.py]: I tried it before. but it has lower performance than `torch.compile(dynamic=True)`

File: vllm_omni/diffusion/layers/rope.py (hunk 2 above)
Comment (SamitHuang) [reply vllm_omni/diffusion/layers/rope.py]: i see. how about NPU. shall we "try except" to avoid failure on NPU?

Comment (ZJY0516) [reply vllm_omni/diffusion/layers/rope.py]: cc @gcanlin

Comment (gcanlin) [reply vllm_omni/diffusion/layers/rope.py]: Could we refer to [this implementation](https://github.com/vllm-project/vllm/blob/fd8afdf38dad8bf7ccc4e7fcc3d4aaa4d6d9e0d8/vllm/model_executor/layers/rotary_embedding/common.py#L121) in vllm? The more details of discussion about it is [here](https://github.com/vllm-project/vllm/pull/29873).

Comment (ZJY0516) [reply vllm_omni/diffusion/layers/rope.py]: Yes, we could. But I don't want to directly use vllm's custom op. We need a seperate mechanism to dispatch

Comment (ZJY0516) [reply vllm_omni/diffusion/layers/rope.py]: Done

**Review by gcanlin — APPROVED** [review]
LGTM, thanks!
